tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php

```diff
@@ -0,0 +1,136 @@
+use SprykerSdk\Sdk\Core\Appplication\Service\CommandExecutor;
+use SprykerSdk\SdkContracts\CommandRunner\CommandRunnerInterface;
+use SprykerSdk\SdkContracts\Entity\CommandInterface;
+use SprykerSdk\SdkContracts\Entity\ContextInterface;
+
+/**
+ * @group Sdk
+ * @group Core
+ * @group Application
+ * @group Service
+ * @group CommandExecutorTest
+ */
+class CommandExecutorTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testExecute(): void
+    {
+        $context = $this->createContextMock();
+        $commandExecutor = new CommandExecutor(
+            [
+                $this->createCommandRunnerMock($context, false),
+                $this->createCommandRunnerMock($context),
+            ],
+            [$this->createAfterCommandExecutedActionMock($context, $this->once())],
+        );
+
+        $result = $commandExecutor->execute($this->createCommandMock(), $context, 'test');
+
+        $this->assertSame($context->getExitCode(), $result->getExitCode());
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecuteIsDry(): void
+    {
+        $context = $this->createContextMock(true);
+        $commandExecutor = new CommandExecutor(
+            [
+                $this->createCommandRunnerMock($context),
+                $this->createCommandRunnerMock($context),
+            ],
+            [$this->createAfterCommandExecutedActionMock($context, $this->never())],
+        );
+
+        $result = $commandExecutor->execute($this->createCommandMock(), $context, 'test');
+
+        $this->assertSame($context->getExitCode(), $result->getExitCode());
+    }
+
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\Entity\CommandInterface
+     */
+    protected function createCommandMock(): CommandInterface
+    {
+        return $this->createMock(CommandInterface::class);
+    }
+
+    /**
+     * @param \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\Entity\ContextInterface $context
+     * @param bool|true $canHandle
+     * @param \PHPUnit\Framework\MockObject\Rule\InvocationOrder|null $invocationRule
+     *
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\CommandRunner\CommandRunnerInterface
+     */
+    protected function createCommandRunnerMock(
+        ContextInterface $context,
+        bool $canHandle = true,
+        ?InvocationOrder $invocationRule = null
+    ): CommandRunnerInterface {
+        if ($invocationRule === null) {
+            $invocationRule = $this->once();
+        }
+        $commandRunner = $this->createMock(CommandRunnerInterface::class);
+        $commandRunner
+            ->expects($invocationRule)
+            ->method('canHandle')
+            ->willReturn($canHandle);
+        $commandRunner
+            ->expects(!$canHandle || $invocationRule->isNever() || $context->isDryRun() ? $this->never() : $this->once())
```

Comment:
Let's keep tests simple, without complicated conditions, etc
Please rewrite it

Reply:
preconditions for expectation. it's unit test